### PR TITLE
Detect negated content absent buffer 2224 v11

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/2224
https://redmine.openinfosecfoundation.org/issues/6629

Describe changes:
- detect: negated content matches on absent buffer
- detect: adds `absent` keyword to match on absent buffer
- doc: fix `byte_test`examples

```
SV_BRANCH=pr/1535
```
https://github.com/OISF/suricata-verify/pull/1535

#10051 with extended commit message

@jufajardini what do you think about it and the doc ?

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17431

File: src/detect-engine.c [hunk 1]
```diff
@@ -691,6 +691,9 @@ static void AppendAppInspectEngine(DetectEngineCtx *de_ctx,
     new_engine->sm_list = t->sm_list;
     new_engine->sm_list_base = t->sm_list_base;
     new_engine->smd = smd;
+    if (t->v2.Callback == DetectEngineInspectBufferGeneric) {
```
Comment (victorjulien) [inline src/detect-engine.c:694]: this means only a subset of sticky buffers can match on this, right? E.g. dns.query or file.data won't be supported, but the rules won't be rejected either?

Comment (catenacyber) [reply src/detect-engine.c:694]: For me dns.query is a multi buffer, not a sticky buffer...
What rule should be rejected ?
What would you expect the behavior to be ?
Currently, negated content on a multi buffer matches if one of the buffer does not have the content, not if all of them do not have the content... right ?
Is there some doc to update ?

I thought `file.data` was a streaming buffer, and not a sticky buffer...
What rule should be rejected ?

Comment (victorjulien) [reply src/detect-engine.c:694]: stickybuffer is just  meant to reflect all the keywords that precede content and friends in the rule syntax, so it includes dns.query, file.data, etc

Comment (catenacyber) [reply src/detect-engine.c:694]: So, I guess the good way to do this for multi-buffer is a rewrite to have `DetectEngineInspectMultiBufferGeneric` which will handle `match_on_null`
And so, to have `InspectionBufferGetDataPtr` augmented by an argument which is the local id of the buffer to get

What do you think ?

Comment (catenacyber) [reply src/detect-engine.c:694]: cf https://redmine.openinfosecfoundation.org/issues/6575

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/10334

**Pull request closed without merging**